src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/LogMessages.cs

```diff
@@ -0,0 +1,74 @@
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+//  Microsoft.IdentityModel.Protocols.Pop
+// SignedHttpRequest range: 23000 - 23999.
+
+namespace Microsoft.IdentityModel.Protocols.SignedHttpRequest
+{
+    /// <summary>
+    /// Log messages and codes
+    /// </summary>
+    internal static class LogMessages
+    {
+        public const string IDX23000 = "IDX23000: CryptoProviderFactory returned null for key: '{0}', signatureAlgorithm: '{1}'.";
+        public const string IDX23001 = "IDX23001: HttpRequestUri must be absolute when creating or validating the 'u' claim. HttpRequestUri: '{0}'.";
+        public const string IDX23002 = "IDX23002: The HTTP Method must be an uppercase HTTP verb. HttpMethod: '{0}'.";
+        public const string IDX23003 = "IDX23003: The signed http request does not contain the '{0}' claim or the claim value is null. This claim is required to validate a signed http request.";
+        public const string IDX23004 = "IDX23004: The following query parameters will not be processed as they are repeated: '{0}'.";
+        public const string IDX23005 = "IDX23005: The following headers will not be processed as they are repeated: '{0}'.";
+        public const string IDX23006 = "IDX23006: The address specified '{0}' is not valid as per the HTTPS scheme. Please specify an HTTPS address for security reasons. For testing with an HTTP address, set the RequireHttpsForJkuResourceRetrieval property on SignedHttpRequestValidationParameters to false.";
+        public const string IDX23007 = "IDX23007: HttpRequestUri is an invalid relative URI: '{0}'.";
+        public const string IDX23008 = "IDX23008: Exception caught while creating the '{0}' claim. Inner exception: '{1}'.";
+        public const string IDX23009 = "IDX23009: Signed http request signature validation failed. Exceptions caught: '{0}'.";
+        public const string IDX23010 = "IDX23010: Lifetime validation of the signed http request failed. Current time: '{0}', signed http request is valid until: '{1}'.";
+        public const string IDX23011 = "IDX23011: '{0}' claim validation failed. Expected value: '{1}', value found: '{2}'.";
+        public const string IDX23012 = "IDX23012: '{0}' claim validation failed. Expected values: '{1}' or '{2}', value found: '{3}'.";
+        public const string IDX23013 = "IDX23013: 'at' token validation failed. Inner exception: '{0}'.";
+        public const string IDX23014 = "IDX23014: Unable to resolve a PoP key. The 'cnf' object must have one of the following claims: 'jwk', 'jwe', 'jku', 'kid'. The 'cnf' claim value: '{0}'.";
+        public const string IDX23015 = "IDX23015: A security key resolved from the 'jwk' claim is not an asymmetric key. Resolved key type: '{0}'.";
+        public const string IDX23016 = "IDX23016: Unable to convert the key found in the 'jwk' claim to a security key. JsonWebKey: '{0}'.";
```

Comment:
will this be needed if we fix up the cnf claim?

Reply:
Still needed. We might get an incorrect JWK that we can't use to create a SecurityKey.

Reply:
todo: resolving pop key might throw.